### ISSUE TITLE
JSDoc support

### ISSUE THREAD

**Issue by richardwillars** [marcj/TypeRunner#9]
Completely agree with the premise of this project, and think it's a great approach you've undertaken. 

One thing I noticed in your notes that you've not included support for JSDoc comments, so I would suggest that this is one that might be reconsidered to be included. There's a lot of libraries out there that use JSDoc comments to describe what functions do, and without them it's unclear how to use them just from the function name and the argument types etc. In VSCode for example, it pulls the JSDoc comments into the intellisense style tooltips, which is incredibly useful.. it just adds an extra layer of context for the developer.

**Comment by marcj** [comment]
> There's a lot of libraries out there that use JSDoc comments to describe what functions do

do you have a list of popular libraries that do use JSDoc instead of d.ts?

> so I would suggest that this is one that might be reconsidered to be included

Very unlikely that TypeRunner will ever support JSDoc.

**Comment by richardwillars** [comment]
First library I looked at was Lodash which uses JSDoc - e.g. https://github.com/lodash/lodash/blob/master/conformsTo.js

**Comment by marcj** [comment]
NPM says it has d.ts files https://www.npmjs.com/package/lodash

**Comment by richardwillars** [comment]
😐 not sure if joking..

**Comment by richardwillars** [comment]
d.ts won't provide the descriptions that you get with JSDoc comments, they serve different purposes when used together. I'm not suggesting using one over the other, I'm suggesting parsing JSDoc comments in order to provide more context in IDEs, not as a type replacement. Aka you have libraries that outputs types and JSDoc comments (like Lodash). Not just libraries either, it's the same for a lot of application code written in house. JSDoc is kinda a big standard used by the majority of people commenting functions etc. Seems kinda naive to just discard it

**Comment by Dimava** [comment]
I think both the Readme and this issue title are badly worded
> JSDoc support
The thing @richardwillars is contextual information about types/variables/arguments written in so-calles "JSDoc comments", `/** comment */` with two opening stars
![image](https://user-images.githubusercontent.com/20068737/197891902-9b56797e-4e5a-4912-ba13-f24250568128.png)

The thing @marcj wants to avoid is type-definition comments, which are *not* supported in TS
![image](https://user-images.githubusercontent.com/20068737/197891310-5610c47e-2246-4be3-9ca1-6c60564d918a.png)
(well, the IDE suggest refactoring to copy types, but that's all. Argument type is `any`.)

I would recommend to switch wording from "JSDoc" to "TSDoc", https://tsdoc.org/ , which is a more strict subset of JSDoc, and does not support type-definitions

**Comment by marcj** [comment]
With JSDoc support this is meant: https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html
So it won't be part of type checking. @richardwillars you screenshot doesn't show anything related to JSDoc.

**Comment by matthew-dean** [comment]
Rich Harris (Rollup / Svelte) moved his smaller libs to JSDoc and [has been an advocate](https://twitter.com/rich_harris/status/1350436286948122625) for using it instead of full TypeScript. See: https://github.com/Rich-Harris/devalue/blob/master/src/parse.js. I've seen other thought leaders in the JS / TS space say the same thing, that maybe JSDoc is all that is needed for public libs. I'm not _personally_ convinced, but there's a strong argument that JSDoc is used quite a bit in TypeScript type-checking.

**Comment by matthew-dean** [comment]
@marcj 

> NPM says it has d.ts files

Yes, exactly. You can do full TypeScript type-checking / compile-time error throwing, `.d.ts` generation using JSDoc only. These are not mutually exclusive. JSDoc is (can be) a full TypeScript environment.
